modules/source-bookshelf/index.js

```diff
@@ -51,6 +62,23 @@ Source.prototype.typeName = function () {
   return this.model.typeName;
 };
 
+// this method should return the related source and a list
+// of ids that should be retreived. the resulting information
+// will be used to call .read on the target source, with a pre-set
+// filter for ids.
+// this is to allow requests like /authors/1/books?filter[thing]=x
+// due to deficiencies in bookshelf, this means we're going to wind
+// up querying for the same data twice because we can't build up a
+// related query and request it in a single go.
+// this will be resolved in a future version of bookshelf
+Source.prototype.related = bPromise.method(function (relation, model) {
+  var relatedModels = relate(model, relation);
+  return [
+    relatedModels.length ? relatedModels.model.__source__ : null,
+    relatedModels.pluck('id')
+  ];
+});
+
```

Comment:
Even with this fix, this method is still erroring out with `[TypeError: Cannot read property 'models' of undefined]`. Not sure where that's coming from